### PR TITLE
byte_extract lowering: lower newly introduced byte_extract expressions [depends-on: #4061, blocks: #2068]

### PR DESCRIPTION
**Pull request by tautschnig**
Only the second commit is new, the first one is #4061.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: unit/solvers/lowering/byte_operators.cpp [hunk 1]
```diff
@@ -154,8 +154,7 @@ SCENARIO("byte_extract_lowering", "[core][solvers][lowering][byte_extract]")
             const exprt lower_be = lower_byte_extract(be, ns);
             const exprt lower_be_s = simplify_expr(lower_be, ns);
 
-            // TODO: does not currently hold
-            // REQUIRE(!has_subexpr(lower_be, ID_byte_extract_little_endian));
+            REQUIRE(!has_subexpr(lower_be, ID_byte_extract_little_endian));
```
Comment (smowton) [inline unit/solvers/lowering/byte_operators.cpp:165]: Presumably also big_endian?

**Review by smowton — APPROVED** [review]
Suspect you need the BE version too, otherwise lgtm

File: unit/solvers/lowering/byte_operators.cpp (hunk 1 above)
Comment (tautschnig) [reply unit/solvers/lowering/byte_operators.cpp:165]: Ah, indeed, actually that should be fixed in the first commit (which is #4061).

Comment (tautschnig) [reply unit/solvers/lowering/byte_operators.cpp:165]: Done.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 36385be05081d365b589f3013d157770c9e1f0c8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99858855

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bae580348666131679114246c4b824696f4ff0c9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100049738

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1f94bceb739eaa8d8bda419d74a961f6a5d7d8dc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100077409

**Pull request merged**